### ISSUE TITLE
Hide longhorn in the legacy apps interface

### ISSUE THREAD

**Issue by MKlimuszka** [rancher/rancher#36176]
In 2.6.3, we have longhorn as an app that can be fresh installed in the old legacy app section. It is the only available app.

We need to hide it starting in 2.6.4 in the old legacy apps UI.

It will still allow “upgrades” to the existing deployed version, which are edits to the version that you are currently running. But we will hide it for fresh installs.

SURE-3812

**Comment by gaktive** [comment]
@Jono-SUSE-Rancher UI transferred this ticket here since it feels like hiding this could be done via existing metadata on the chart.

**Comment by innobead** [comment]
cc @kaxing @meldafrawi

**Comment by PennyScissors** [comment]
The master branch in the rancher/charts repository is public facing, so this issue needs to be tested after pull request https://github.com/rancher/charts/pull/1747 is approved, but before it is merged. 

### What changed
All longhorn charts in the legacy catalog will be hidden. This was done by adding a rancher max version constraint equal to 2.6.3 so starting from 2.6.4 the longhorn charts should be hidden in the legacy catalog.

### What is affected
All longhorn charts in the legacy catalog (`rancher/charts` repo, `master` branch).

### How to test it
1. Get a Rancher setup of version greater than 2.6.3 (latest 2.6.4 RC will do the trick)
2. Enable legacy flag in `Global Settings > Feature Flags`
3. In the top right, filter by a project (if local cluster, you can use `Project: Default`)
4. Navigate to the cluster's `Legacy > Catalogs` and add the catalog using URL `https://github.com/pennyscissors/charts`, and branch `m-hide-longhorn` (this will bring the changes from my pull request)
5. Navigate to `Legacy > Projects > Apps > Launch`, and verify none of the longhorn charts are shown under the name of the new catalog you added.

**Comment by innobead** [comment]
@khushboo-rancher @kaxing we need to ensure the upgrade still working in this case if the user already has a longhorn version installed.

**Comment by khushboo-rancher** [comment]
@PennyScissors I tried with catalog `https://github.com/pennyscissors/charts`, and branch m-hide-longhorn, I can still see the Longhorn app getting listed in Legacy App.

Steps taken:
1. Created a Rancher with v2.6.4-rc4 chart with v2.6-head `93f3e13` image
2. Enabled the Legacy flag
3. Added the catalog with name `test-longhorn`
<img width="1551" alt="Screen Shot 2022-03-02 at 2 48 46 PM" src="https://user-images.githubusercontent.com/60111667/156463877-be060629-fe91-407f-8da1-300424c37557.png">
4. Navigated to `Legacy > Projects > Apps > Launch`, I could see the Longhorn app to install
<img width="1547" alt="Screen Shot 2022-03-02 at 2 52 37 PM" src="https://user-images.githubusercontent.com/60111667/156463972-eeaf3e9d-869e-4502-b2ff-f8101c48672f.png">

**Comment by PennyScissors** [comment]
@innobead Can you elaborate on the upgrade case you want @khushboo-rancher to test exactly? If a user installs longhorn 1.2.2 in Rancher 2.6.3, then the current behavior is preserved and they can upgrade to a newer longhorn version. If the user upgrades to Rancher 2.6.4, then the rancher version constraint kicks in and the expected behavior is they can only modify their current installation, but they can't upgrade to a newer version within the legacy catalog or re-install their chart because the idea is we want them to use the charts in Apps & Marketplace instead (This is how all other hidden charts work currently). So by upgrade path, are you referring to the case when the user edits their current chart installation in Rancher 2.6.4? (This is seen as an upgrade to the same chart version with new values in the backend, and should work).

**Comment by innobead** [comment]
> I looked back in the PR and I think I may have misunderstood [this](https://github.com/rancher/charts/pull/1747#issuecomment-1048643658) question. It does not affect upgrading to the same version for the purpose of editing the values of your existing installation, however, it does prevent users from upgrading to a newer version of the legacy longhorn chart. There is no logic in rancher that will selectively allow users to upgrade their existing installation to a newer version that has been purposely hidden if they have an existing installation.

What @khushboo-rancher tested is correct, I would expect we hide the Longhorn app in the legacy catalog for fresh install only, but users who already installed the Longhorn app should be able to upgrade to newer versions.

**Comment by PennyScissors** [comment]
@innobead There is no way to prevent fresh installs without preventing upgrades to newer versions in the current logic to hide charts in the legacy catalog. This would be a feature that need to be implemented in the legacy catalog and we aren't taking any more feature requests for it because it has been replaces by the Apps & Marketplace.

**Comment by innobead** [comment]
> @innobead There is no way to prevent fresh installs without preventing upgrades to newer versions in the current logic to hide charts in the legacy catalog. ~This would be a feature that need to be implemented in the legacy catalog and we aren't taking any more feature requests for it because it has been replaced by the Apps & Marketplace~. Actually, this would become a UI task since it will need to check if a longhorn legacy chart is installed, and if one is installed, then show newer longhorn versions regardless of being hidden. The rancher version constraint still needs to remain in the chart to prevent fresh installs.

Thanks, @PennyScissors for the detailed explanation. 

We probably need to postpone this change from the upcoming 2.6.4 until we figure out first the migration path for the users who already installed the existing Longhorn version from the catalog.

cc @MKlimuszka @SheilaghM @deniseschannon @wjimenez5271

**Comment by PennyScissors** [comment]
@innobead No problem! I'll go a head and revert my PR. Let me know if you need any help with the migration path.

**Comment by PennyScissors** [comment]
@innobead actually, no need to revert the PR, it was never merged because master is a public facing branch and we wanted to test it before it is merged/published. So you can consider this resolved for the moment. I'll leave the PR open in case we need it in the future.

**Comment by SheilaghM** [comment]
This ticket is on hold waiting on work from the Longhorn team.

**Comment by MKlimuszka** [comment]
This was finished via UI
